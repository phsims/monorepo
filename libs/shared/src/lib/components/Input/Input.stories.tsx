import type { Meta, StoryObj } from '@storybook/react';
import { Input } from './Input';

import { within } from '@storybook/testing-library';
import { expect } from '@storybook/jest';

const meta: Meta<typeof Input> = {
  component: Input,
  title: 'Input',
};
export default meta;
type Story = StoryObj<typeof Input>;

export const Primary = {
  args: {
    id: 'text-input',
    lable: 'input example',
    name: 'input',
    placeholder: 'text input example',
  },
};

export const Heading: Story = {
  args: {},
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    expect(canvas.getByText(/Welcome to Input!/gi)).toBeTruthy();
  },
};
