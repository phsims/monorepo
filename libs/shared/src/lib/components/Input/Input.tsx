/* eslint-disable-next-line */
export interface InputProps {
  id: string;
  lable: string;
  name: string;
  type?: string;
  required?: boolean;
  placeholder?: string;
}

export function Input({
  id,
  lable,
  name,
  type = 'text',
  required = false,
  placeholder,
}: InputProps) {
  return (
    <div>
      <label htmlFor={id} className="sr-only">
        {lable}
      </label>
      <input
        id={id}
        name={name}
        type={type}
        autoComplete={type}
        required={required}
        className="relative block w-full appearance-none rounded-md  border border-lightgrey border-opacity-40 px-3 py-2 text-gray-900 placeholder-gray-500 focus:z-10 focus:border-indigo-500 focus:outline-none focus:ring-indigo-500 sm:text-sm"
        placeholder={placeholder || lable}
      />
    </div>
  );
}

export default Input;
