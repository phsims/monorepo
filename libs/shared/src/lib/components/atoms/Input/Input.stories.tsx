import type { Meta, StoryObj } from '@storybook/react';
/* eslint-disable-next-line */
import { Markdown } from '@storybook/blocks';
import { within } from '@storybook/testing-library';
import { expect } from '@storybook/jest';

import ReadMe from './README.md?raw';
import { Input } from './Input';

const meta: Meta<typeof Input> = {
  component: Input,
  title: '@shared/atoms/Input',
  tags: ['autodocs'],
  parameters: {
    docs: {
      description: {
        component: `<Markdown>${ReadMe}</Markdown>`,
      },
    },
  },
  decorators: [
    (Story) => (
      <div className="max-w-md">
        <Story />
      </div>
    ),
  ],
};
export default meta;
type Story = StoryObj<typeof Input>;

export const Default: Story = {
  args: {
    id: 'text-input',
    lable: 'input example',
    name: 'input',
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    expect(canvas.getByText(/input example/gi)).toBeTruthy();
  },
};
