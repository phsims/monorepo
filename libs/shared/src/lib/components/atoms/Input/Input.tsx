import React from 'react';

import { kebabCase } from '../../../utilities/text-helpers';

export interface InputProps {
  label: string;
  id?: string;
  name?: string;
  type?: string;
  required?: boolean;
  placeholder?: string;
  className?: string;
  autoComplete?: string;
  aria?: string;
}

export function Input({
  label,
  placeholder,
  autoComplete,
  name = kebabCase(label),
  id = kebabCase(label),
  type = 'text',
  required = false,
  aria = label,
  className = 'relative block w-full appearance-none rounded-md  border border-lightgrey border-opacity-40 px-3 py-2 text-gray-900 placeholder-gray-500 focus:z-10 focus:border-indigo-500 focus:outline-none focus:ring-indigo-500 sm:text-sm',
}: InputProps) {
  return (
    <>
      <label htmlFor={id} className="sr-only">
        {label}
      </label>
      <input
        id={id}
        name={name}
        type={type}
        required={required}
        className={className}
        placeholder={placeholder || label}
        autoComplete={autoComplete}
        aria-label={aria}
      />
    </>
  );
}

export default Input;
