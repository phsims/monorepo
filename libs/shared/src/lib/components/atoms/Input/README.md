<!-- README.md -->



Primary UI component for user interaction

```js
import { Input } from "@shared";
```