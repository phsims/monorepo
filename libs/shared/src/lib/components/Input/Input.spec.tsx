import { render } from '@testing-library/react';

import Input from './Input';
const props = {
  id: 'text-input',
  lable: 'input example',
  name: 'input',
  placeholder: 'text input example',
};
describe('Input', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Input {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
