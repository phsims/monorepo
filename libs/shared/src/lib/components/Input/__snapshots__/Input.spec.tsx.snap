// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render successfully 1`] = `
<body>
  <div>
    <div>
      <label
        class="sr-only"
        for="text-input"
      >
        input example
      </label>
      <input
        autocomplete="text"
        class="relative block w-full appearance-none rounded-md  border border-lightgrey border-opacity-40 px-3 py-2 text-gray-900 placeholder-gray-500 focus:z-10 focus:border-indigo-500 focus:outline-none focus:ring-indigo-500 sm:text-sm"
        id="text-input"
        name="input"
        placeholder="text input example"
        type="text"
      />
    </div>
  </div>
</body>
`;
